File: app/src/main/java/fergaral/datetophoto/utils/TickedImageView.java
package fergaral.datetophoto.utils;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.support.annotation.NonNull;
import android.util.AttributeSet;
import android.view.View;
import android.widget.ImageView;
import fergaral.datetophoto.R;

/**
 * Created by fer on 24/05/15.
 */
public class TickedImageView extends ImageView {

    private boolean checked;
    private Bitmap mTickBmp;
    private Paint mDarkerPaint;
    private View.OnClickListener onImageClickListener;
    private int x, y;

    public TickedImageView(Context context) {
        super(context);
        init();
    }

    public TickedImageView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    private void init() {
        super.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                setChecked(!isChecked());

                if (onImageClickListener != null)
                    onImageClickListener.onClick(TickedImageView.this);
            }
        });

        mDarkerPaint = new Paint(Paint.ANTI_ALIAS_FLAG);
        mDarkerPaint.setStyle(Paint.Style.FILL);
        mDarkerPaint.setColor(0x80142030);

        mTickBmp = BitmapFactory.decodeResource(getResources(), R.drawable.ic_done_white_48px);
    }

    public void setChecked(boolean checked) {
        this.checked = checked;
        invalidate();
    }

    public boolean isChecked() {
        return checked;
    }

    @Override
    protected void onDraw(@NonNull Canvas canvas) {
        super.onDraw(canvas);

        if (checked) {
            canvas.drawRect(0, 0, canvas.getWidth(), getMeasuredWidth(), mDarkerPaint);
            canvas.drawBitmap(mTickBmp, x, y, null);
        }
    }

    @Override
    public void setOnClickListener(OnClickListener listener) {
        onImageClickListener = listener;
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, widthMeasureSpec);

        x = (getMeasuredWidth() / 2) - (mTickBmp.getWidth() / 2);
        y = (getMeasuredWidth() / 2) - (mTickBmp.getHeight() / 2);
    }
}

File: app/src/main/java/fergaral/datetophoto/utils/TickedImageView.kt
package fergaral.datetophoto.utils

import android.content.Context
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.graphics.Canvas
import android.graphics.Paint
import android.util.AttributeSet
import android.view.View
import android.widget.ImageView
import fergaral.datetophoto.R

/**
 * Created by fer on 24/05/15.
 */
class TickedImageView : ImageView {

    var isChecked: Boolean = false
        set(checked) {
            field = checked
            invalidate()
        }
    private var mTickBmp: Bitmap? = null
    private var mDarkerPaint: Paint? = null
    private var onImageClickListener: View.OnClickListener? = null
    private var x: Int = 0
    private var y: Int = 0

    constructor(context: Context) : super(context) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init()
    }

    private fun init() {
        super.setOnClickListener {
            isChecked = !isChecked

            if (onImageClickListener != null)
                onImageClickListener!!.onClick(this@TickedImageView)
        }

        mDarkerPaint = Paint(Paint.ANTI_ALIAS_FLAG)
        mDarkerPaint!!.style = Paint.Style.FILL
        mDarkerPaint!!.color = -0x7febdfd0

        mTickBmp = BitmapFactory.decodeResource(resources, R.drawable.ic_done_white_48px)
    }

    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)

        if (isChecked) {
            canvas.drawRect(0f, 0f, canvas.width.toFloat(), measuredWidth.toFloat(), mDarkerPaint!!)
            canvas.drawBitmap(mTickBmp!!, x.toFloat(), y.toFloat(), null)
        }
    }

    override fun setOnClickListener(listener: View.OnClickListener?) {
        onImageClickListener = listener
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, widthMeasureSpec)

        x = measuredWidth / 2 - mTickBmp!!.width / 2
        y = measuredWidth / 2 - mTickBmp!!.height / 2
    }
}
